### PR TITLE
Add AdvancedTextMiddleware unit tests

### PR DESCRIPTION
**Pull request by cl445**
## Was

Dedizierte Unit-Tests für `AdvancedTextMiddleware`, das bisher nur indirekt über die volle Pipeline angelaufen wurde.

**Coverage: 29 % → 100 %** (alle 8 Funktionen).

## Abgedeckt

- **Delete-forward** — Löschen bzw. No-op je nach Folgekontext (leer / nil)
- **Capitalize-word** — Groß-/Kleinschreiben, Wortgrenze stoppt an Leerzeichen und Ziffern, deutsches Locale (`ß → SS`), No-op bei leerem Kontext / Nicht-Buchstabe
- **Clipboard** — copy/paste/cut inkl. der `hasFullAccess`-Guards

## Hinweise

- Genutzt wird die vorhandene `MockTextTarget`-Infrastruktur (kein Simulator-Keyboard nötig).
- Die Clipboard-Suite ist `@Suite(.serialized)`, da `UIPasteboard.general` ein Prozess-Singleton ist und Swift Testing parallel läuft.

## Testlauf

20 neue Tests grün auf `iPhone 17 Pro`.

## Summary by CodeRabbit

* **Tests**
  * Added coverage for advanced text editing behavior, including forwarding, delete-forward, word capitalization, and clipboard actions.
  * Verified edge cases such as empty text, cursor boundaries, non-letter characters, and locale-specific capitalization.
  * Confirmed clipboard actions only work with the required access and that copy/paste/cut behave correctly without unintended document changes.

### CODE REVIEW

File: wurstfingerTests/AdvancedTextMiddlewareTests.swift [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+        let target = MockTextTarget()
+        target.hasFullAccess = true
+        target.selectedText = nil
+        let middleware = AdvancedTextFixtures.middleware(target: target)
+
+        middleware.process(AdvancedTextFixtures.context(.cut)) { _ in }
+
+        #expect(target.events.isEmpty)
+    }
+}
```
[the remark this answers is not in the document]
Comment (cl445) [reply wurstfingerTests/AdvancedTextMiddlewareTests.swift:303]: Übernommen: Suite ist jetzt eine `final class`, die `UIPasteboard.general` in `init` sichert und in `deinit` wiederherstellt.

File: wurstfingerTests/AdvancedTextMiddlewareTests.swift [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+    @Test func cutIsNoopWhenNothingSelected() {
+        let target = MockTextTarget()
+        target.hasFullAccess = true
+        target.selectedText = nil
+        let middleware = AdvancedTextFixtures.middleware(target: target)
+
+        middleware.process(AdvancedTextFixtures.context(.cut)) { _ in }
+
+        #expect(target.events.isEmpty)
```
[the remark this answers is not in the document]
Comment (cl445) [reply wurstfingerTests/AdvancedTextMiddlewareTests.swift:300]: Übernommen: Beide Cut-No-op-Tests seeden einen Marker und prüfen, dass das Pasteboard unverändert bleibt.
